### ISSUE TITLE
the graph function is not equal to the original function

### ISSUE THREAD

**Issue by wc411421** [tulip-control/dd#49]
The function I want to expressed in the dd package is "**!x & y & z**", and the following is my code.
```python
from dd import autoref as _bdd

bdd = _bdd.BDD()
bdd.declare('x', 'y', 'z')

u = bdd.add_expr('(!x) & y & z')
bdd.collect_garbage()
bdd.dump('a.png')
```

I get the following graphic:
![a](https://user-images.githubusercontent.com/24862284/77497091-6c9b9f80-6e87-11ea-9885-7325ce0d6c7c.png)

According to the rules, I deduce the function "**x | !x & !(y&z)**" from the graph function, and it is not equal to the original function **"!x & y & z**".
What did I do wrong during this process?

**Comment by johnyf** [comment]
External references to BDDs can be negated when using the class `dd.autoref.BDD` (and also the classes `dd.cudd.BDD`, `dd.sylvan.BDD`, `dd.bdd.BDD`). What this means is that the reference `u` in this case is itself a negated edge:

```python
>>> u = bdd.add_expr('(!x) & y & z')
>>> print(u.negated)
True
```

The reference `u` is a complemented arc that points to the node `x-7` in the diagram. This can be confirmed by printing the integer representation of the node `u` (which is how the classes `dd.autoref.Function` and `dd.autoref.BDD` are implemented, by the class `dd.bdd.BDD`):

```python
>>> print(u.node)
-7
```

The external reference can be included in the diagram by passing the argument `roots` to the method `dd.autoref.BDD.dump`:

```python
from dd import autoref as _bdd

bdd = _bdd.BDD()
bdd.declare('x', 'y', 'z')

u = bdd.add_expr('(!x) & y & z')
bdd.collect_garbage()
bdd.dump('a.png', roots=[u])
```

The result is the following diagram, where the node `@-7` is the external reference `u`:

![a](https://user-images.githubusercontent.com/3135273/77535445-4b48ac80-6ea3-11ea-8b20-791f0003db4b.png)

Another way to confirm that the expression `x | !x & !(y&z)` is the negation of the expression `!x & y & z` is:

```python
u = bdd.add_expr('(!x) & y & z')
v = bdd.add_expr('x | !x & !(y&z)')
assert u == ~ v
```

Relevant to #39, https://github.com/tulip-control/dd/issues/29#issuecomment-289612426, #41.

**Comment by wc411421** [comment]
Thanks for your reply. I think the concept of **external reference** is very important, but I only find it mentioned in the document when talking about the attributes of node u in **Other method**.
![db47ce8482b47b3b6fb06cf104bacf1](https://user-images.githubusercontent.com/24862284/77602528-b4363000-6f48-11ea-8610-d1426fc0e869.png)

I recommend reflecting the **external reference** when introducing the **Plotting** chapter.

**Comment by johnyf** [comment]
Addressed in e032bb639514a3ce4cc0d7b056bb91056b121457.
